src/include/dng/task.h

```diff
@@ -44,5 +72,108 @@ class Task {
 
 } // namespace dng
 
+namespace dng {
+namespace task {
+// Support the use of bool_switch_t and tribool in boost::program_options
+inline
+void validate(boost::any &v, const std::vector<std::string> &xs,
+              bool_switch_t *, int) {
+    boost::program_options::validate(v, xs, (bool*)0, 0);
+    bool b = boost::any_cast<bool>(v);
+    v = boost::any(bool_switch_t{b});
+}
+
+#if !defined(BOOST_NO_STD_WSTRING)
+inline
+void validate(boost::any &v, const std::vector<std::wstring> &xs,
+              bool_switch_t *, int) {
+    boost::program_options::validate(v, xs, (bool*)0, 0);
+    bool b = boost::any_cast<bool>(v);
+    v = boost::any(bool_switch_t{b});
+}
+#endif
+}
+}
+
+namespace boost {
+inline
+void validate(boost::any &v, const std::vector<std::string> &xs,
+              boost::tribool *, int) {
+    using namespace boost::program_options;
+    validators::check_first_occurrence(v);
+    std::string s(validators::get_single_string(xs, true));
+
+    for(size_t i = 0; i < s.size(); ++i) {
+        s[i] = char(tolower(s[i]));
+    }
+
+    if(s.empty() || s == "on" || s == "yes" || s == "1" || s == "true") {
+        v = boost::any(boost::tribool(true));
+    } else if(s == "off" || s == "no" || s == "0" || s == "false") {
+        v = boost::any(boost::tribool(false));
```

Comment:
This code is designed to be compatible with boost::program_options's `bool` validation, which doesn't support `n` and `y`.